orchestra/bots/tests/test_staffbot.py

```diff
@@ -208,3 +218,41 @@ def test_restaff_command_errors(self):
         self.assertEquals(response.get('text'),
                           (bot.task_assignment_does_not_exist_error
                            .format(worker.user.username, task.id)))
+
+    def test_get_staffing_request_messsage(self):
+        def _task_factory(status, path):
+            description_no_kwargs = {'path': path}
+            return TaskFactory(
+                status=status,
+                step=StepFactory(
+                    slug='stepslug',
+                    description='the step',
+                    detailed_description_function=description_no_kwargs),
+                project=ProjectFactory(
+                    workflow_version=WorkflowVersionFactory(
+                        workflow=WorkflowFactory(description='the workflow'))))
+
+        task = _task_factory(
+            Task.Status.AWAITING_PROCESSING,
+            'orchestra.tests.helpers.fixtures.get_detailed_description')
+        staffing_request_inquiry = StaffingRequestInquiryFactory(
+            request=StaffBotRequestFactory(
+                task=task))
+        message = StaffBot()._get_staffing_request_message(
+            staffing_request_inquiry,
+            'communication/new_task_available_slack.txt')
+
+        self.assertEqual(message,
+                         "Hello!\n\nA new task is available for you to work on, if you'd like!  Here are the details:\n\nProject type: the workflow\nTask type: the step\nMore details: No text given stepslug\n\n<http://127.0.0.1:8000/orchestra/communication/accept_staffing_request_inquiry/{}/|Accept the Task>\n<http://127.0.0.1:8000/orchestra/communication/reject_staffing_request_inquiry/{}/|Reject the Task>\n\n".format(staffing_request_inquiry.id, staffing_request_inquiry.id))  # noqa
```

Comment:
it took me quite a while to get this working.  i'm not particularly excited to change it, unless we think it significantly improves things for us :)